### PR TITLE
Fix runtest

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Summary

Fixes GHA tests and adds a way to specify a custom stanc3 binary (required if changes are made in Stan/Math and stanc3 in sync, for example https://github.com/stan-dev/stan/pull/3011).

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Rok Češnovar

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -135,6 +135,9 @@ pipeline {
                        git clean -xffd
                     """
                     utils.checkout_pr("math", "lib/stan_math", params.math_pr)
+                    if (params.stanc3_bin_url != 'nightly') {
```
Comment (rok-cesnovar) [inline Jenkinsfile:138]: @serban-nicusor-toptal can you add a textbox in the Build with parameters for `stanc3_bin_url`.

**Comment by serban-nicusor-toptal** [comment]
That should do it @rok-cesnovar. I'm not sure if defaultValue and description is correct tho.

**Comment by rok-cesnovar** [comment]
@serban-nicusor-toptal thanks!

mind reviewing when you have the chance? No rush.

**Comment by rok-cesnovar** [comment]
Refactored this a bit, ready for review now.

**Review by serban-nicusor-toptal — APPROVED** [review]
LGTM!

**Pull request merged**